ngraph/core/CMakeLists.txt

```diff
@@ -72,6 +72,11 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
     set_target_properties(ngraph PROPERTIES LINK_FLAGS "/IGNORE:4217,4286")
 endif()
 
+set_source_files_properties("${CMAKE_CURRENT_SOURCE_DIR}/src/pass/convert_precision.cpp"
+                            "${CMAKE_CURRENT_SOURCE_DIR}/src/pass/convert_fp32_to_fp16.cpp"
+                            "${CMAKE_CURRENT_SOURCE_DIR}/src/op/type_relaxed.cpp"
+        PROPERTIES INCLUDE_DIRECTORIES "${CMAKE_CURRENT_SOURCE_DIR}/../../inference-engine/src/transformations/include/")
```

Comment:
I really dislike the idea to use a hack with relative paths in order to use headers from the third-party library without explicit dependency

Reply:
As GraphRewrite inherits FunctionPass and has the same API so there is no any backward compatibility issues.

Reply:
Explicit dependency nGraph->InferenceEngineTransformations leads to cycle dependency issue. So this is the only way how we can use code from InferenceEngineTransformations library inside nGraph transformations.
_@ilya-lavrenov could you please comment on this?_

Reply:
I dislike the relative pathes as well. The recommendation is to use `$<TARGET_PROPERTY:inference_engine_transformations,INTERFACE_INCLUDE_DIRECTORIES>`

@ilyachur If you have concerns about the approach itself, let's discuss online